### PR TITLE
[WIP] Removing CreateFileW from UWP

### PR DESCRIPTION
**Pull request by pjanotti**
First pass at removing all CreateFileW calls in UWP

### CODE REVIEW

File: src/System.IO.Ports/src/System.IO.Ports.csproj [hunk 1]
```diff
@@ -85,7 +85,7 @@
     <Compile Include="$(CommonPath)\Interop\Windows\kernel32\Interop.WaitCommEvent.cs">
       <Link>Common\Interop\Windows\kernel32\Interop.WaitCommEvent.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\Interop\Windows\kernel32\Interop.CreateFile.cs">
+    <Compile Condition="'$(TargetGroup)' != 'uap'" Include="$(CommonPath)\Interop\Windows\kernel32\Interop.CreateFile.cs">
```
Comment (danmosemsft) [inline src/System.IO.Ports/src/System.IO.Ports.csproj]: can yo uput the condition on an itemgroup? seems to be what is generally done

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateNamedPipeClient.Uap.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (danmosemsft) [inline src/Common/src/Interop/Windows/kernel32/Interop.CreateNamedPipeClient.Uap.cs]: how is this file any different than the existing Interop.UnsafeCreatefile.Uap.cs ? can we use that

Comment (pjanotti) [reply src/Common/src/Interop/Windows/kernel32/Interop.CreateNamedPipeClient.Uap.cs]: The problem is this one gets out as SafePipelineHandle, at first try I can't cast it from the UnsafeCreateFile.Uap - it was my first try, let me know if there is a way to do that.

File: src/System.IO.Ports/src/System.IO.Ports.csproj (hunk 1 above)
Comment (pjanotti) [reply src/System.IO.Ports/src/System.IO.Ports.csproj]: done

**Comment by danmosemsft** [comment]
Looks like you can also delete these
```
C:\git\corefx\src\System.Runtime.Extensions\src\PinvokeAnalyzerExceptionList.analyzerdata.uap:
    1: shell32.dll!SHGetKnownFolderPath

C:\git\corefx\src\System.Runtime.Extensions\src\PinvokeAnalyzerExceptionList.analyzerdata.uapaot:
    1: shell32.dll!SHGetKnownFolderPath
```

**Comment by pjanotti** [comment]
Checking in a local build before pushing the removal of the 2 files under System.Runtime.Extensions

File: src/Common/src/Interop/Windows/kernel32/Interop.CreateNamedPipeClient.Uap.cs (hunk 2 above)
Comment (danmosemsft) [reply src/Common/src/Interop/Windows/kernel32/Interop.CreateNamedPipeClient.Uap.cs]: Uh, yeah I guess this is the best I can think of.

File: src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs [hunk 3]
```diff
@@ -27,7 +27,7 @@ private void StartRaisingEvents()
 
             // Create handle to directory being monitored
             var defaultSecAttrs = default(Interop.Kernel32.SECURITY_ATTRIBUTES);
-            _directoryHandle = Interop.Kernel32.CreateFile(
+            _directoryHandle = Interop.Kernel32.UnsafeCreateFile(
```
Comment (danmosemsft) [inline src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs]: why this change?

Comment (pjanotti) [reply src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs]: UnsafeCreateFile is the wrapper used to redirect between Win32 and Uap implementation. This is the approach used in System.IO.FileSystem

Comment (danmosemsft) [reply src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs]: oh, it's to abstract createfile/createfile2. I was confused because of the name of this file. the file should be FileSystemWatcher.Windows.cs, I think? Win32 means not UAP.

Comment (pjanotti) [reply src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs]: Good point.

**Comment by pjanotti** [comment]
@dotnet-bot test this please
